builder.go

```diff
@@ -5,75 +5,100 @@ import (
 	"io"
 	"io/ioutil"
 	"os"
-	"os/exec"
 	"path/filepath"
 	"sort"
 
 	"github.com/BurntSushi/toml"
-	"github.com/pkg/errors"
 
 	"github.com/buildpacks/lifecycle/api"
 	"github.com/buildpacks/lifecycle/launch"
 	"github.com/buildpacks/lifecycle/layers"
 )
 
-type Builder struct {
-	AppDir        string
-	LayersDir     string
-	PlatformDir   string
-	BuildpacksDir string
-	PlatformAPI   *api.Version
-	Env           BuildEnv
-	Group         BuildpackGroup
-	Plan          BuildPlan
-	Out, Err      io.Writer
-}
-
 type BuildEnv interface {
 	AddRootDir(baseDir string) error
 	AddEnvDir(envDir string) error
 	WithPlatform(platformDir string) ([]string, error)
 	List() []string
 }
 
-type LaunchTOML struct {
-	Labels    []Label
-	Processes []launch.Process `toml:"processes"`
-	Slices    []layers.Slice   `toml:"slices"`
+type BuildpackFinder interface {
+	Find(bpID, bpVersion, buildpacksDir string) (BuildpackTOML, error)
 }
 
-type Label struct {
-	Key   string `toml:"key"`
-	Value string `toml:"value"`
+type DefaultBuildpackFinder struct{}
+
+// TODO: this duplicates code in buildpack.go
+func (f *DefaultBuildpackFinder) Find(bpID, bpVersion, buildpacksDir string) (BuildpackTOML, error) {
```

Comment:
Ah, I tried calling the interface `BuildpackStore` but metadata.go already has a `BuildpackStore` struct - maybe this could become `BuildpackStoreTOML`?

Reply:
Trying out `BuildpackStore` -> `StoreTOML` for the existing struct. Then we can use `BuildpackStore` and `DirBuildpackStore` here.